### PR TITLE
[NNAPI EP] Update NNAPI headers

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Update the NNAPI headers to a more recent version (copied from TF Lite v2.9.1).

**Motivation and Context**
Enables use of newer NNAPI functionality such as the ANEURALNETWORKS_BATCH_MATMUL operation.

### CODE REVIEW

File: onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_lib/nnapi_implementation.cc [hunk 1]
```diff
@@ -201,10 +201,14 @@ const NnApi LoadNnApi() {
   LOAD_FUNCTION(libneuralnetworks, ANeuralNetworksModel_finish);
   LOAD_FUNCTION(libneuralnetworks, ANeuralNetworksModel_addOperand);
   LOAD_FUNCTION(libneuralnetworks, ANeuralNetworksModel_setOperandValue);
-  LOAD_FUNCTION_OPTIONAL(libneuralnetworks, ANeuralNetworksModel_setOperandSymmPerChannelQuantParams);
-  LOAD_FUNCTION(libneuralnetworks, ANeuralNetworksModel_setOperandValueFromMemory);
+  LOAD_FUNCTION_OPTIONAL(
```
Comment (edgchen1) [inline onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_lib/nnapi_implementation.cc:204]: some formatting changes were made to minimize the diff between the TF Lite version of this file

File: onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_lib/NeuralNetworksWrapper.h [hunk 2]
```diff
@@ -38,9 +36,7 @@ enum class Type {
   TENSOR_QUANT16_SYMM = ANEURALNETWORKS_TENSOR_QUANT16_SYMM,
   TENSOR_FLOAT16 = ANEURALNETWORKS_TENSOR_FLOAT16,
   TENSOR_BOOL8 = ANEURALNETWORKS_TENSOR_BOOL8,
-  FLOAT16 = ANEURALNETWORKS_FLOAT16,
   TENSOR_QUANT8_SYMM_PER_CHANNEL = ANEURALNETWORKS_TENSOR_QUANT8_SYMM_PER_CHANNEL,
-  TENSOR_QUANT16_ASYMM = ANEURALNETWORKS_TENSOR_QUANT16_ASYMM,
```
Comment (YUNQIUGUO) [inline onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_lib/NeuralNetworksWrapper.h:43]: why are we removing `TENSOR_QUANT16_ASYMM`/`FLOAT16` types here? 
is it to align with TF lite latest version

Comment (edgchen1) [reply onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_lib/NeuralNetworksWrapper.h:43]: yes, it matches this:
https://github.com/tensorflow/tensorflow/blob/d8ce9f9c301d021a69953134185ab728c1c248d3/tensorflow/lite/nnapi/NeuralNetworksTypes.h#L37-L51

File: onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_lib/nnapi_implementation_stub.cc [hunk 3]
```diff
@@ -0,0 +1,245 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (edgchen1) [inline onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_lib/nnapi_implementation_stub.cc:1]: move stubs into a separate file to minimize nnapi_implementation.cc diff with TF Lite

**Review by YUNQIUGUO — APPROVED** [review]
Looks good to me.

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**